h2o-automl/src/test/java/hex/ModelTest.java

```diff
@@ -0,0 +1,94 @@
+package hex;
+
+import ai.h2o.automl.targetencoding.BlendingParams;
+import ai.h2o.automl.targetencoding.TargetEncoder;
+import hex.genmodel.ModelDescriptor;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import water.Key;
+import water.TestUtil;
+import water.fvec.Frame;
+
+import java.util.Map;
+
+/**
+ * This test should be moved to h2o-core module once we move all TargetEncoding related classes there as well. 
+ */
+public class ModelTest extends TestUtil implements ModelTestCommons{
+
+  @BeforeClass public static void stall() { stall_till_cloudsize(1); }
+
+
+  private Map<String, Frame> getTEMap() {
+    Frame trainFrame = parse_test_file("./smalldata/gbm_test/titanic.csv");
+    String responseColumnName = "survived";
+    asFactor(trainFrame, responseColumnName);
+    
+    BlendingParams params = new BlendingParams(3, 1);
+    String[] teColumns = {"home.dest", "embarked"};
+    TargetEncoder targetEncoder = new TargetEncoder(teColumns, params);
+    Map<String, Frame> testEncodingMap = targetEncoder.prepareEncodingMap(trainFrame, responseColumnName, null);
+    return testEncodingMap;
+  }
+  
+  @Test public void addTargetEncodingMap() {
+    TestModel.TestParam p = new TestModel.TestParam();
+    TestModel.TestOutput o = new TestModel.TestOutput();
+    TestModel testModel = new TestModel(Key.make(),p,o);
+    Map<String, Frame> teMap = getTEMap();
+    testModel.addTargetEncodingMap(teMap);
+
+    checkEncodings(testModel._output._target_encoding_map);
+    //TODO key leakage
```

Comment:
todo

Reply:
@honzasterba Thanks, going to fix all the leakages as I'm I guess at the final stage of the PR